src/gateway/shard.rs

```diff
@@ -121,13 +122,15 @@ impl Shard {
     /// TLS error.
     pub async fn new(
         ws_url: Arc<Mutex<String>>,
+        ws_proxy: Arc<Mutex<Option<String>>>,
```

Comment:
[...] and this are breaking changes. Would you mind if you retargeted the PR to the `next` branch?